lib/srv/db/mysql/engine.go

```diff
@@ -191,6 +197,26 @@ func (e *Engine) connect(ctx context.Context, sessionCtx *common.Session) (*clie
 		if err != nil {
 			return nil, trace.Wrap(err)
 		}
+		// workaround issue generating ephemeral certs for secure connections
+		// by creating a TLS connection to the cloud proxy port (3307) and
+		// overridding mysql client's connection. mysql on 3306 does not trust
+		// the ephemeral cert's CA but cloud proxy does.
+		if sessionCtx.Database.GetTLS().Mode != types.DatabaseTLSMode_INSECURE {
+			uri := sessionCtx.Database.GetURI()
+			host, port, err := net.SplitHostPort(uri)
+			if err == nil && port == "3306" {
+				uri = net.JoinHostPort(host, "3307")
+				e.Log.Debug("Overrided URI port from 3306 to 3307")
+			}
+			tlsconn, err := tls.Dial("tcp", uri, tlsConfig)
+			if err != nil {
+				return nil, trace.Wrap(err)
+			}
+			clientOpt = func(conn *client.Conn) {
+				conn.Conn.Close() // close connection created by mysql client
```

Comment:
Letting the library dial the connection we know we don't need and then closing it feels a bit hacky TBH. There's [ConnectWithDialer](https://github.com/gravitational/go-mysql/blob/master/client/conn.go#L71) method we can probably use instead.